src/NHibernate.Test/NHSpecificTest/NH3023/DeadlockHelper.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Data.SqlClient;
+using System.Threading;
+using System.Transactions;
+using log4net;
+
+namespace NHibernate.Test.NHSpecificTest.NH3023
+{
+	public partial class DeadlockHelper
+	{
+		private static readonly ILog _log = LogManager.GetLogger(typeof(DeadlockHelper));
+
+		public void ForceDeadlockOnConnection(SqlConnection connection)
+		{
+			using (var victimLock = new SemaphoreSlim(0))
+			using (var winnerLock = new SemaphoreSlim(0))
+			{
+				//
+				// Second thread with non-pooled connection, to deadlock
+				// with current thread
+				//
+				Exception winnerEx = null;
+				var winnerThread = new Thread(
+					() =>
+					{
+						try
+						{
+							using (var scope = new TransactionScope(TransactionScopeOption.RequiresNew))
```

Comment:
Initial version was `TransactionScopeOption.Requires` instead of `TransactionScopeOption.RequiresNew`. Changed for async test sake: with `TransactionScopeAsyncFlowOption.Enabled`, the transaction from the test thread was propagated to the created thread, breaking the test logic.